### ISSUE TITLE
Malisis door compat screws with textures/lighting

### ISSUE THREAD

**Issue by praryboy** [AbrarSyed/SecretRoomsMod-forge#176]
Minecraft version - 1.7.10
SecretRoomsMod version - 1.7.10-4.7.1.409
MinecraftForge version - 10.13.2.1291

Trying out the newest version, found a few issues. For one, any secret door that was already on the map before the update has to be broken and replaced before it will work.

But more importantly, the new doors and trapdoors have mirrored textures when they are facing north or east. Also the lighting is bizarre, reminiscent of forge microblock "covers", where it is very obvious there is something hidden there. This didn't happen before and still doesn't happen with blocks like the ghost block. Im assuming the malisis door is some kind of entity and its lighting is handled differently.

Any way to go back to the normal secret doors while still having malisis doors running? 

http://i.imgur.com/3mq0d7S.png

**Comment by AbrarSyed** [comment]
@Ordinastie 

The replacing of the doors I know is an issue.. and there is not a whole lot I can do about it in my estimation.. so far atleast. The mirroring.. well.. poked ordinastie about that..

**Comment by praryboy** [comment]
Don't forget the lighting, haha. When you have a nice stone brick wall, but two blocks look like they have a shadow on them, it's pretty obvious something is amiss. Forge microblocks have had this problem since Eloraam first made them, and as far as I understand it's because of their entity status, it's probably very hard to make them render exactly like a block. Of course, I haven't tried it myself.

I'd gladly give up animated doors for properly hidden ones, if this can't be fixed easily.

**Comment by AbrarSyed** [comment]
yeah One thing I probably can do is just add a config option to disable the compat thingy...

**Comment by Ordinastie** [comment]
Fixing the flipped texture right now, but for the lighting issue, I would need more details, or rather, an explicit example of the conditions leading to wrong lighting.
The renderer do handle the lighthing and shading for TileEntities the same way it should be handled for block, but I admit for SecretRooms, there are some special cases that need to be taken care of.

**Comment by praryboy** [comment]
Well it happens for me on every single door and trapdoor, although it's less noticeable with some textures. If you want a specific, verifiable instance, have a look at this picture
http://imgur.com/5Efv6KJ
on the floor you can see 4 trapdoors pretty clearly. Each one is facing a different direction, so unlike the texture mirror that doesn't seem to make a difference.

the right wall is just a plain wall, the lighting is even across it. The left wall has a secret door to the left of the torch. You can see the lighting is different on both pieces of the door. The wall on the other side of the door is also darker, as if trying to move across the door absorbed some of the light levels (this may just be an optical illusion though).

http://imgur.com/xQ4f3Bj

Here you can see what happens when you put the torch directly in front of the door (left) vs in front of a normal wall (right). It just doesn't seem to matter where the lights are in relation to the door, the lighting on it makes it stand out. 

The testing conditions are simple. A small room carved out underground from smooth stone. The only light sources within 30 blocks are the torches you can see in the picture.

**Comment by AbrarSyed** [comment]
flipping should be fixed thanks to #177

**Comment by praryboy** [comment]
That's awesome. I'd love to have that config option you mentioned before too.

**Comment by AbrarSyed** [comment]
config option is there already.https://github.com/AbrarSyed/SecretRoomsMod-forge/commit/b005a33b8bd914db8687bd1fc52efd233487160d

**Comment by praryboy** [comment]
phenomenal
